Packs/GuardiCore/Integrations/AkamaiGuardicoreEventCollector/AkamaiGuardicoreEventCollector.py

```diff
@@ -0,0 +1,264 @@
+        context = {"access_token": access_token,
+                   "expires_in": expiration_timestamp.strftime(DATE_FORMAT)}
+        set_integration_context(context)
+        demisto.debug(
+            f"New access token that expires in : {expiration_timestamp.strftime(DATE_FORMAT)}"
+            f" was set to integration_context.")
+
+    def authenticate(self):
+        body = {
+            'username': self.username,
+            'password': self.password
+        }
+        new_token = self._http_request(
+            method='POST',
+            url_suffix='/authenticate',
+            json_data=body)
+
+        if not new_token or not new_token.get('access_token'):
+            raise DemistoException(
+                f"{INTEGRATION_NAME} error: The client credentials are invalid.")
+
+        new_token = new_token.get('access_token')
+        return new_token
+
+    def get_events(self, start_time, end_time, limit, offset) -> List[Dict[str, Any]]:
+        """
+        Get events from Guardicore API using the modelbreaches endpoint and the start and end time.
+        """
+        params = {
+            "from_time": start_time,
+            "to_time": end_time,
+            "limit": limit,
+            "offset": offset
+        }
+        print(params)
+        data = self._http_request(
+            method='GET',
+            url_suffix=f'/incidents',
+            params=params
+        )
+        return data['objects']
+
+
+def get_jwt_expiration(token: str):
+    if "." not in token:
+        return 0
+    jwt_token = base64.b64decode(token.split(".")[1] + '==')
+    jwt_token = json.loads(jwt_token)
+    return jwt_token.get("exp")
+
+
+def generate_new_token(self):
+    token = self.authenticate()
+    self.save_jwt_token(token)
+    self._save_access_token(token)
+
+
+def test_module(client: Client, params: Dict[str, Any]) -> str:
+    """
+    Tests API connectivity and authentication'
+    When 'ok' is returned it indicates the integration works like it is supposed to and connection to the service is
+    successful.
+    Raises exceptions if something goes wrong.
+    Args:
+        client (Client): Gurdicore client to use.
+        params (Dict): Integration parameters.
+    Returns:
+        str: 'ok' if test passed, anything else will raise an exception and will fail the test.
+    """
+
+    try:
+        client.login()
+
+    except Exception as e:
+        if 'UNAUTHORIZED' in str(e):
+            return 'Authorization Error: make sure the username and password are correctly set'
+        else:
+            raise e
+
+    return 'ok'
+
+
+def add_time_to_events(events):
+    """ Adds the _time key to the events.
+
+    Args:
+        events: list[dict] - list of events to add the _time key to.
+
+    Returns:
+        list: The events with the _time key.
+    """
+    if events:
+        for event in events:
+            event['_time'] = event.get('start_time')
```

Comment:
It should be faster to do this in comprehension i.e.
```suggestion
        [{**event, '_time': event.get('start_time')} for event in events]
```